front/src/applications/editor/components/EditorForm.tsx

```diff
@@ -84,6 +84,16 @@ function EditorForm<T extends Omit<EditorEntity, 'objType'> & { objType: string
     }
   }, [error]);
 
+  useEffect(() => {
+    if (!data.properties.curves || !data.properties.slopes) {
+      setFormData({
+        ...data.properties,
+        slopes: data.properties.slopes ?? [],
+        curves: data.properties.curves ?? [],
+      });
+    }
+  }, [data.properties.length]);
```

Comment:
You don't need this, the best thing to do I think is to initialize curves and slopes to [] in getInitialState (see getNewLine, add curves: [] and slopes: []).
You will need to add curves and slopes to the TrackSectionEntity
```
      slopes: LinearMetadataItem<{ gradient: number }>[];
      curves: LinearMetadataItem<{ radius: number }>[];
```
So you can remove the useEffect
```suggestion
```